### PR TITLE
feat: reschedule interface in risedev 

### PR DESCRIPTION
**Pull request by shanicky**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

This PR provides the `risedev ctl meta reschedule` interface, which allows you to call the reschedule api of meta directly from the command line.

Use the `--plan` parameter to provide a reschedule plan, which can provide multiple fragment at the same time  separated by `;`, and the `-dry-run` parameter will only show the parsed results without operation

```
100-[1,2,3]+[4,5];101-[1];102+[3]
```
For fragment 100,  parallel units 1,2,3 will be removed, while parallel units 4,5 will be added. 

You can also provide only the removed parallel unit ids (like fragment 101) or only the added pu ids (like fragment 102), but in the case of both, the removed must precede the added

The parsed result:
```
For fragment #100
	Remove: [1, 2, 3]
	Add:    [4, 5]
For fragment #102
	Add:    [3]
For fragment #101
	Remove: [1]
```

## Checklist

~- [ ] I have written necessary rustdoc comments~
~- [ ] I have added necessary unit tests and integration tests~
~- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)~

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/ctl/src/cmd_impl/meta/reschedule.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+use std::collections::HashMap;
+
+use anyhow::{anyhow, Error, Result};
+use regex::{Match, Regex};
+use risingwave_pb::meta::reschedule_request::Reschedule;
+
+use crate::common::MetaServiceOpts;
+
+const RESCHEDULE_MATCH_REGEXP: &str =
+    r"^(?P<fragment>\d+)(?:-\[(?P<removed>\d+(?:,\d+)*)])?(?:\+\[(?P<added>\d+(?:,\d+)*)])?$";
```
Comment (yezizp2012) [inline src/ctl/src/cmd_impl/meta/reschedule.rs:24]: 🧎🏻
          🧎🏻🧎🏻
     🧎🏻🧎🏻🧎🏻
🧎🏻🧎🏻🧎🏻🧎🏻
🧎🏻🧎🏻🧎🏻🧎🏻
     🧎🏻🧎🏻🧎🏻
          🧎🏻🧎🏻
               🧎🏻

Comment (BugenZhao) [reply src/ctl/src/cmd_impl/meta/reschedule.rs:24]: 🧎🏻🧎🏻🧎🏻🧎🏻
🧎🏻🧎🏻🧎🏻🧎🏻
🧎🏻🧎🏻🧎🏻
🧎🏻🧎🏻
🧎🏻

File: src/ctl/src/lib.rs [hunk 2]
```diff
@@ -113,6 +113,15 @@ enum MetaCommands {
     Resume,
     /// get cluster info
     ClusterInfo,
+    /// Reschedule the parallel unit in the stream graph
+    Reschedule {
+        /// Plan of reschedule
+        #[clap(long)]
+        plan: String,
```
Comment (MrCroxx) [inline src/ctl/src/lib.rs:120]: Better add an example here.

**Review by MrCroxx — COMMENTED** [review]
LGTM! Success!

File: src/ctl/src/cmd_impl/meta/reschedule.rs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+const RESCHEDULE_FRAGMENT_KEY: &str = "fragment";
+const RESCHEDULE_REMOVED_KEY: &str = "removed";
+const RESCHEDULE_ADDED_KEY: &str = "added";
+
+pub async fn reschedule(plan: String, dry_run: bool) -> Result<()> {
```
Comment (yezizp2012) [inline src/ctl/src/cmd_impl/meta/reschedule.rs:29]: Please add an example and expected output for this function.

File: src/ctl/src/lib.rs (hunk 2 above)
Comment (BugenZhao) [reply src/ctl/src/lib.rs:120]: +1, which will be used for generating the command line helps automatically.

**Review by BugenZhao — COMMENTED** [review]
LSTM!!!!!

**Review by yezizp2012 — APPROVED** [review]
LSSSSTM!

**Review by BugenZhao — APPROVED** [review]

**Comment by shanicky** [comment]
The help message has been added, and executing `risedev ctl meta reschedule --help` will show this to output
```Reschedule the parallel unit in the stream graph

The format is `fragment_id-[removed]+[added]`
You can provide either `removed` only or `added` only, but `removed` should be preceded by
`added` when both are provided.

For example, for plan `100-[1,2,3]+[4,5]` the follow request will be generated:
{
    100: Reschedule {
        added_parallel_units: [4,5],
        removed_parallel_units: [1,2,3],
    }
}
Use ; to separate multiple fragment

USAGE:
    risectl meta reschedule [OPTIONS] --plan <PLAN>

OPTIONS:
        --dry-run
            Show the plan only, no actual operation

    -h, --help
            Print help information

        --plan <PLAN>
            Plan of reschedule

    -V, --version
            Print version information```

File: src/ctl/src/cmd_impl/meta/reschedule.rs (hunk 1 above)
Comment (wyhyhyhyh) [reply src/ctl/src/cmd_impl/meta/reschedule.rs:24]: 🧎🏻🧎🏻🧎🏻🧎🏻🧎🏻

Comment (wyhyhyhyh) [reply src/ctl/src/cmd_impl/meta/reschedule.rs:24]: 🧎🏻🧎🏻
   🧎🏻🧎🏻
   🧎🏻🧎🏻
 🧎🏻🧎🏻
🧎🏻🧎🏻🧎🏻🧎🏻

Comment (wyhyhyhyh) [reply src/ctl/src/cmd_impl/meta/reschedule.rs:24]:
```
   🧎🏻🧎🏻🧎🏻🧎🏻🧎🏻         🧎🏻🧎🏻🧎🏻🧎🏻🧎🏻   
         🧎🏻                          🧎🏻
         🧎🏻                         🧎🏻
   🧎🏻🧎🏻🧎🏻🧎🏻🧎🏻         🧎🏻🧎🏻🧎🏻🧎🏻🧎🏻  
         🧎🏻                      🧎🏻
         🧎🏻                   🧎🏻🧎🏻
   🧎🏻🧎🏻🧎🏻🧎🏻🧎🏻               🧎🏻
```

**Pull request merged**